newIDE/app/src/InstancesEditor/index.js

```diff
@@ -804,16 +811,72 @@ export default class InstancesEditor extends Component<Props> {
     this.viewPosition.scrollTo(x, y);
   }
 
-  centerView() {
+  zoomToFitContent() {
+    const instanceMeasurer = this.instancesRenderer.getInstanceMeasurer();
+    let contentAABB: ?Rectangle;
+    const getInstanceRectangle = new gd.InitialInstanceJSFunctor();
+    // $FlowFixMe - invoke is not writable
+    getInstanceRectangle.invoke = instancePtr => {
+      // $FlowFixMe - wrapPointer is not exposed
+      const instance = gd.wrapPointer(instancePtr, gd.InitialInstance);
+      if (!contentAABB) {
+        contentAABB = instanceMeasurer.getInstanceAABB(
+          instance,
+          new Rectangle()
+        );
+      } else {
+        contentAABB.union(
+          instanceMeasurer.getInstanceAABB(instance, new Rectangle())
+        );
+      }
+    };
+    const { initialInstances } = this.props;
+    if (initialInstances.getInstancesCount() > 0) {
+      // $FlowFixMe - JSFunctor is incompatible with Functor
+      initialInstances.iterateOverInstances(getInstanceRectangle);
+
+      if (contentAABB) {
+        const idealZoom = this.viewPosition.fitToRectangle(contentAABB);
+        this.setZoomFactor(idealZoom);
+      }
+    }
+  }
+
+  zoomToInitialPosition() {
```

Comment:
Is this the correct initial position?
If the user change the game start position, does it also change the origin of the canvas? (0,0) (maybe I'm just imagining they can)

Reply:
I don't think it's possible to change the origin of the scene.